ifrs17-template/Test/ReimportWithDifferentScope.ipynb

```diff
@@ -0,0 +1,198 @@
+                "\n    .ExecuteAsync()"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var ifrsVars1 = await DataSource.Query<IfrsVariable>().ToArrayAsync();",
+                "\nvar dn1 = ifrsVars1.Select(x => x.DataNode).ToHashSet();",
+                "\nvar dt11Prem1 = ifrsVars1.Where(x => x.DataNode == \"DT1.1\" && x.AmountType == \"PR\");",
+                "\n(ifrsVars1.Count(), dn1.Count(), dt11Prem1.Count())"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Cash flow Import2",
+                "\nImporting a file for the same period as previous import but it only contains 1 GIC: DT1.1 where one amount type (Premium) has been removed"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "await Import.FromFile(\"Data/NominalCashflows_CH_2020_12_DT1.1NoPrem.csv\")",
+                "\n    .WithFormat(ImportFormats.Cashflow)",
+                "\n    .WithTarget(DataSource)",
+                "\n    .WithActivityLog(Session, DataSource, DataSetReader)",
+                "\n    .ExecuteAsync()"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "var ifrsVars2 = await DataSource.Query<IfrsVariable>().ToArrayAsync();",
+                "\nvar dn2 = ifrsVars2.Select(x => x.DataNode).ToHashSet();",
+                "\nvar dt11Prem2 = ifrsVars2.Where(x => x.DataNode == \"DT1.1\" && x.AmountType == \"PR\");",
+                "\n(ifrsVars2.Count(), dn2.Count(), dt11Prem2.Count())"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "# Test"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Import generate variables"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "ifrsVars1.Count().Should().NotBe(0);"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "ifrsVars2.Count().Should().NotBe(0);"
```

Comment:
This test can be made stricter. Since in the 2nd import no premiums are parsed, perhaps ifrsVars2 must be less ifrsVars1.

Reply:
well technically you don't know a priory what happens with CSM/LC. You might have several switches that then result in more variables.